### ISSUE TITLE
OutputLabel.for would not evaluate target correctly

### ISSUE THREAD

**Issue by javaserverfaces** [javaee/mojarra#4176]
I talked about this problem here [http://stackoverflow.com/questions/38608286/poutputlabel-not-marked-as-required-when-target-input-is-inside-composite](http://stackoverflow.com/questions/38608286/poutputlabel-not-marked-as-required-when-target-input-is-inside-composite) and there [https://github.com/primefaces/primefaces/issues/1633](https://github.com/primefaces/primefaces/issues/1633). The problem is not from Primefaces. I just tried with native components like h:outputLabel and h:inputText.

Basically, after getting component referenced by "for" attribute, the component defined as required is not evaluated as such...
#### Affected Versions
[2.2.13, 2.3.0-m06]

**Comment by edburns** [comment]
Please see this important message regarding community contributions to
Mojarra.

https://javaee.groups.io/g/jsf-spec/message/30

Also, please consider joining that group, as that group has taken the
place of the old dev@javaserverfaces.java.net mailing list.

Thanks,

Ed Burns
